http/server_test.ts

```diff
@@ -364,6 +364,11 @@ test(async function testReadRequestError(): Promise<void> {
       in: "HEAD / HTTP/1.1\r\nContent-Length:0\r\nContent-Length: 0\r\n\r\n",
       headers: [{ key: "Content-Length", value: "0" }],
       err: null
+    },
+    11: {
+      in: "ToTAlly \\ Non REAdable HTTTTTTp pRotOcOlz",
+      err: "EOF",
+      headers: []
```

Comment:
This test doesn't demonstrate the error. When I run it without your other change, it passes.

Reply:
Mm you're right. We may work on a way to mock the request flow in the test runner. I'll try to find a way.

Reply:
Have you looked at these tests? https://github.com/golang/go/blob/master/src/net/http/requestwrite_test.go

They're quite nice. I think we need to add ResponseWriter.

Reply:
In this case it will not be revelant because we will not write the response. But yes it would be a good addition sure. Adding this in another PR?